packages/react/src/table/TableScroller.tsx

```diff
@@ -0,0 +1,336 @@
+
+		setButtonIntervalId(intervalId);
+	};
+
+	const handleButtonRelease = () => {
+		if (buttonIntervalId) {
+			clearInterval(buttonIntervalId);
+			setButtonIntervalId(null);
+		}
+	};
+
+	const handleTrackClick = (event: React.MouseEvent) => {
+		if (!scrollerRef?.current || !thumbRef?.current) {
+			return;
+		}
+
+		const thumbDimensions = thumbRef.current.getBoundingClientRect();
+
+		if (event.pageX > thumbDimensions.x + thumbDimensions.width) {
+			scrollerRef.current.scrollLeft += thumbDimensions.width * 0.95;
+		} else if (event.pageX < thumbDimensions.x) {
+			scrollerRef.current.scrollLeft -= thumbDimensions.width * 0.95;
+		}
+	};
+
+	return (
+		<Stack
+			gap={0.5}
+			css={{
+				position: 'relative',
+				width: '100%',
+			}}
+		>
+			<Box
+				css={{
+					msOverflowStyle: 'none',
+					overflowX: 'auto',
+					scrollbarWidth: 'none',
+					WebkitOverflowScrolling: 'touch',
+					width: '100%',
+					'&::-webkit-scrollbar, &::-webkit-scrollbar-thumb, &::-webkit-scrollbar-track':
+						{
+							display: 'none',
+						},
+				}}
+				focusRingFor="keyboard"
+				onScroll={repositionThumb}
+				ref={scrollerRef}
+				tabIndex={0}
+			>
+				{children}
+			</Box>
+			<Flex
+				background="body"
+				alignItems="center"
+				css={{
+					bottom: 0,
+					display: thumbWidthRatio === 1 ? 'none' : undefined,
+					left: 0,
+					position: 'sticky',
+					right: 0,
+				}}
+				flexWrap="nowrap"
+				gap={0.25}
+			>
+				<Box
+					aria-hidden
+					as="button"
+					css={{
+						appearance: 'none',
+						background: 'none',
+						cursor: 'default',
+						height: 24,
+						width: 24,
+					}}
+					onClick={() => handleButtonClick('left')}
+					onMouseDown={(event: React.MouseEvent) =>
+						handleButtonPress(event, 'left')
+					}
+					onMouseUp={handleButtonRelease}
+					onMouseLeave={handleButtonRelease}
+					onTouchStart={(event: React.TouchEvent) =>
+						handleButtonPress(event, 'left')
+					}
+					onTouchEnd={handleButtonRelease}
+					tabIndex={-1}
+					type="button"
+				>
+					<ScrollbarArrowLeftIcon color="border" />
+				</Box>
+				<Box
+					aria-hidden
+					background="shade"
+					border
+					borderColor="muted"
+					css={{
+						borderRadius: 999,
+						height: 12,
+						padding: 0,
+						position: 'relative',
+						flexGrow: 1,
+					}}
+					onClick={handleTrackClick}
+					ref={trackRef}
+					tabIndex={-1}
+				>
+					<Box
+						aria-hidden
+						as="button"
+						css={{
+							appearance: 'none',
+							background: boxPalette.border,
+							border: 'none',
+							borderRadius: 999,
+							bottom: 0,
+							cursor: 'default',
+							left: thumbPosition,
+							padding: 0,
+							position: 'absolute',
+							top: 0,
+							touchAction: 'none', // Prevent default touch actions
+							width: `${thumbWidthRatio * 100}%`,
+						}}
+						onMouseDown={handleThumbPress}
+						onTouchStart={handleThumbPress}
+						ref={thumbRef}
+						tabIndex={-1}
+						type="button"
```

Comment:
I'm curious what this does?

Reply:
The `type="button"`? Purely to stop the html validator in tests complaining